provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/GradleKotlinDslIntegrationTest.kt

```diff
@@ -421,9 +421,58 @@ class GradleKotlinDslIntegrationTest : AbstractIntegrationTest() {
 
         """)
 
-       assertThat(
-           build("help").output,
-           containsString("*abc*"))
+        assertThat(
+            build("help").output,
+            containsString("*abc*"))
+    }
+
+    @Test
+    fun `settings script can use buildscript dependencies`() {
```

Comment:
:x: We need a similar test for the model builder proving it returns the correct classpath when settings has a `buildscript` block.

Reply:
Good point!
Coverage added